src/app/routes/connector-ui/contract-agreement-page/contract-agreement-page/contract-agreement-page.service.ts

```diff
@@ -43,30 +46,49 @@ export class ContractAgreementPageService {
   }
 
   private fetchData(): Observable<Fetched<ContractAgreementPageData>> {
-    return this.edcApiService.getContractAgreementPage().pipe(
-      Fetched.wrap({failureMessage: 'Failed fetching contract definitions'}),
-      Fetched.map((contractAgreementPage) =>
-        this.buildContractAgreementPageData(contractAgreementPage),
+    return combineLatest([
+      this.edcApiService.getContractAgreementPage(),
+      this.fetchLimits(),
+    ]).pipe(
+      map(([contractAgreementPage, connectorLimits]) =>
+        this.buildContractAgreementPageData(
+          contractAgreementPage,
+          connectorLimits,
+        ),
       ),
+      Fetched.wrap({failureMessage: 'Failed fetching Contract Agreement Page'}),
     );
   }
 
   private buildContractAgreementPageData(
     contractAgreementPage: ContractAgreementPage,
+    connectorLimits: ConnectorLimits | null,
   ): ContractAgreementPageData {
     let contractAgreements = this.mapContractAgreements(
       contractAgreementPage.contractAgreements,
     );
+
+    let consumingContractAgreements = contractAgreements.filter(
+      (it) => it.direction === 'CONSUMING',
+    );
+
+    if (connectorLimits?.maxActiveConsumingContractAgreements != null) {
+      consumingContractAgreements.map((it) => {
+        it.isActivated =
+          consumingContractAgreements.indexOf(it) <
```

Comment:
indexOf has bad performance, and is an antipattern.

you can simply use `.map((it, index) => {` on js arrays